test/suites/basic.sh

```diff
@@ -41,6 +41,44 @@ test_interactive() {
     validate_system_lxd "${m}" 3 disk1
   done
 
+  # Reset the systems with just LXD and no IPv6 support.
+  reset_systems 3 3 1
+
+  for m in micro01 micro02 micro03 ; do
+    lxc exec "${m}" -- echo 1 > /proc/sys/net/ipv6/conf/all/disable_ipv6
+    lxc exec "${m}" -- snap disable microceph || true
+    lxc exec "${m}" -- snap disable microovn || true
+    lxc exec "${m}" -- snap restart microcloud
+  done
+
+  echo "Creating a MicroCloud with ZFS storage and no IPv6 support"
+  microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init > out"
+
+  lxc exec micro01 -- tail -1 out | grep "MicroCloud is ready" -q
+  for m in micro01 micro02 micro03 ; do
+    validate_system_lxd "${m}" 3 disk1
+  done
+
+  # Reset the systems with just LXD and no IPv4 support.
+  gw_net_addr=$(lxc network get lxdbr0 ipv4.address)
+  lxc network set lxdbr0 ipv4.address none
+  reset_systems 3 3 1
+
+  for m in micro01 micro02 micro03 ; do
+    lxc exec "${m}" -- snap disable microceph || true
+    lxc exec "${m}" -- snap disable microovn || true
+    lxc exec "${m}" -- snap restart microcloud
+  done
+
+  echo "Creating a MicroCloud with ZFS storage and no IPv4 support"
+  ! microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init 2> out" || false
+
+  # Ensure we exited with the right error message.
+  lxc exec micro01 -- tail -1 err | grep "Error: Could not determine Fan overlay subnet: No default gateway for IPv4" -q
```

Comment:
```suggestion
  lxc exec micro01 -- tail -1 out | grep "Error: Could not determine Fan overlay subnet: No default gateway for IPv4" -q
```

My diagnosis: too much go error handling for you ;)

Reply:
fixed